client/app/states/templates/editor/editor.state.js

```diff
@@ -0,0 +1,40 @@
+/* eslint camelcase: "off" */
+
+/** @ngInject */
+export function TemplatesEditorState(routerHelper) {
+  routerHelper.configureStates(getStates());
+}
+
+function getStates() {
+  return {
+    'templates.editor': {
+      url: '/edit/:templateId',
+      templateUrl: 'app/states/templates/editor/editor.html',
+      controller: StateController,
+      controllerAs: 'vm',
+      resolve: {
+        template: resolveTemplate,
+      },
+    },
+  };
+}
+
+/** @ngInject */
+function resolveTemplate($stateParams, TemplatesService) {
+  if ($stateParams.templateId) {
```

Comment:
was thinking more along the lines of :hocho:ing the two `returns`

`return $state.templateId? TemplatesService.getTemplate($stateParams.templateId) : {};